### PR TITLE
Add PerformerDetector, a system for detecting when performers enter a specified region

### PR DESCRIPTION
**Pull request by azeey**
The `PerformerDetector` system publishes a message when a performer enters or leaves its region. This can be used as a trigger in another system to do something in response to a performer approaching an area.

The system does not assume that levels are enabled, but it does need performers to be specified.
I needed to change the code in `LevelManager.cc` to read in performer info even when levers are not enabled.

I'll keep this as a draft while I work on a system that spawns objects by using the PerformerDetector as a trigger.

### CODE REVIEW

File: examples/worlds/performer_detector.sdf [hunk 1]
```diff
@@ -0,0 +1,494 @@
+<?xml version="1.0" ?>
+
+<!--
+  Try sending commands:
+
+    To move the vehicle:
+
+      ign topic -t "/model/vehicle_blue/cmd_vel" -m ignition.msgs.Twist -p "linear: {x: 0.5}"
```
Comment (chapulina) [inline examples/worlds/performer_detector.sdf:8]: How about also letting them know to echo the detector topic?

`ign topic -e -t /performer_detector`

File: src/systems/performer_detector/PerformerDetector.cc [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+void PerformerDetector::PostUpdate(
+  const ignition::gazebo::UpdateInfo &_info,
+  const ignition::gazebo::EntityComponentManager &_ecm)
+{
+  IGN_PROFILE("PerformerDetector::PostUpdate");
+
+  if (_info.paused)
+    return;
+
+  if (this->initialized)
```
Comment (chapulina) [inline src/systems/performer_detector/PerformerDetector.cc:104]: Nitpick: you could return early if not initialized. This should reduce the indentation of the code below and IMO make it easier to read.

File: src/systems/performer_detector/PerformerDetector.cc [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+  {
+    auto modelPose =
+        _ecm.Component<components::Pose>(this->model.Entity())->Data();
+
+    // Double negative because AxisAlignedBox does not currently have operator+
+    // that takes a position
```
Comment (chapulina) [inline src/systems/performer_detector/PerformerDetector.cc:110]: :scream: https://github.com/ignitionrobotics/ign-math/issues/121

File: src/systems/performer_detector/PerformerDetector.hh [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+namespace ignition
+{
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace systems
+{
+
+  class IGNITION_GAZEBO_VISIBLE PerformerDetector
```
Comment (chapulina) [inline src/systems/performer_detector/PerformerDetector.hh:39]: Mind adding some documentation? :innocent:  Important to mention whether the detection happens on intersection or full containment, and what the published message means.

**Comment by nkoenig** [comment]
> This plugin is very similar to Gazebo-classic's [ContainPlugin](https://github.com/osrf/gazebo/blob/gazebo11/plugins/ContainPlugin.hh), did you get a chance to look at that? Some differences being:
> 
> * The usage of performers
> * The usage of `AxisAlignedBox` instead of `OrientedBox` (if I remember correctly, this was a helpful difference during the SRC)
> * The use of `Intersects` instead of `Contains`
> * The message being published
> * The "enabled" feature
> 
> It would be nice if we could say that the contain plugin has been migrated to Ignition. Do you think we could at least make the change to oriented box?

Those are good suggestions. The problem is timing. We need this out the door asap in order to meet a deadline on Wednesday next week. Can these features be saved for future improvements, maybe captured in an issue and moved on the project board?

**Comment by chapulina** [comment]
> Can these features be saved for future improvements, maybe captured in an issue and moved on the project board?

Sounds good, if @azeey doesn't get to it I can add it to my list of issues to ticket. I think the missing features can be added in a backwards-compatible way.

**Comment by azeey** [comment]
> This plugin is very similar to Gazebo-classic's [ContainPlugin](https://github.com/osrf/gazebo/blob/gazebo11/plugins/ContainPlugin.hh), did you get a chance to look at that? Some differences being:

Yes, @iche033 mentioned that in our design discussion. I didn't know that the `ContainPlugin` existed and I had already written this system by that point. I agree we can add the missing features in a backward-compatible way. The only issue I see is the name. I could rename this to EntityDetector, but  that would give a false impression that this would work with any entity. Maybe we can rename the system later and use ign-plugin name aliases?

**Comment by iche033** [comment]
yeah there are also a couple of other plugins in gazebo classic do something similar so it would be nice to make this generic. But if we don't have any plans / time to commit to this in the near future, we probably just do what @azeey suggetesd and rename and create an alias later when the time comes.

**Comment by chapulina** [comment]
Sure, I don't think the name is a big deal, we've renamed other things as we ported them. We just need a migration guide for each plugin :slightly_smiling_face: 

> EntityDetector, but that would give a false impression that this would work with any entity.

Given that the entity needs a geometry, and to be movable, I think limiting the plugin to performers is reasonable.

File: src/systems/performer_detector/PerformerDetector.hh (hunk 4 above)
Comment (azeey) [reply src/systems/performer_detector/PerformerDetector.hh:71]: https://github.com/ignitionrobotics/ign-gazebo/pull/125/commits/5d4efb6bdd8cc7f4de86d99a52e3d5f0b6a83e93

File: src/systems/performer_detector/PerformerDetector.cc (hunk 3 above)
Comment (azeey) [reply src/systems/performer_detector/PerformerDetector.cc]: Thanks for creating the issue

File: src/systems/performer_detector/PerformerDetector.cc (hunk 2 above)
Comment (azeey) [reply src/systems/performer_detector/PerformerDetector.cc]: Done. https://github.com/ignitionrobotics/ign-gazebo/pull/125/commits/728969a2fc588a2cf834988ddf456e7618a298f8

File: examples/worlds/performer_detector.sdf (hunk 1 above)
Comment (azeey) [reply examples/worlds/performer_detector.sdf:8]: https://github.com/ignitionrobotics/ign-gazebo/pull/125/commits/728969a2fc588a2cf834988ddf456e7618a298f8

**Pull request merged**